sys/net/gnrc/network_layer/ipv6/nib/_nib-internal.h

```diff
@@ -0,0 +1,476 @@
+
+/**
+ * @brief   Gets a default router by IPv6 address and interface
+ *
+ * @pre     `(router_addr != NULL)`
+ *
+ * @param[in] router_addr   The address of a default router. May not be NULL.
+ * @param[in] iface         The interface to the node. May be 0 for any interface.
+ *
+ * @return  The NIB entry for node with @p router_addr and @p iface onsuccess.
+ * @return  NULL, if there is no such entry.
+ */
+_nib_dr_t *_nib_drl_get(const ipv6_addr_t *router_addr, unsigned iface);
+
+/**
+ * @brief   Gets *the* default router.
+ *
+ * @see [RFC 4861, section 6.3.6](https://tools.ietf.org/html/rfc4861#section-6.3.6)
+ *
+ * @return  The current default router, on success.
+ * @return  NULL, if there is no default router in the list.
+ */
+_nib_dr_t *_nib_drl_get_dr(void);
+
+/**
+ * @brief   Creates or gets an existing off-link entry by next hop and prefix
+ *
+ * @pre `(next_hop != NULL)`
+ * @pre `(pfx != NULL) && (pfx != "::") && (pfx_len != 0) && (pfx_len <= 128)`
+ *
+ * @param[in] next_hop  An IPv6 address to next hop. May not be NULL.
+ *                      *May also be a global address!*
+ * @param[in] iface     The interface to @p next_hop.
+ * @param[in] pfx       The IPv6 prefix or address of the destinaion.
+ *                      May not be NULL or unspecified address. Use
+ *                      @ref _nib_drl_add() for default route destinations.
+ * @param[in] pfx_len   The length in bits of @p pfx in bits.
+ *
+ * @return  A new or existing off-link entry with _nib_dr_t::pfx set to @p pfx.
+ * @return  NULL, if no space is left.
+ */
+_nib_dst_t *_nib_dst_alloc(const ipv6_addr_t *next_hop, unsigned iface,
+                           const ipv6_addr_t *pfx, unsigned pfx_len);
+
+/**
+ * @brief   Clears out a NIB entry (off-link version)
+ *
+ * @param[in] dst  An entry
+ */
+void _nib_dst_clear(_nib_dst_t *dst);
+
+/**
+ * @brief   Iterates over off-link entries
+ *
+ * @param[in] last  Last entry (NULL to start)
+ *
+ * @return  entry after @p last
+ */
+_nib_dst_t *_nib_dst_iter(const _nib_dst_t *last);
+
+_nib_dst_t *_nib_dc_add(const ipv6_addr_t *next_hop, unsigned iface,
+                        const ipv6_addr_t *dst);
+
+static inline void _nib_dc_remove(_nib_dst_t *nib_dst)
+{
+    _nib_t *nib = nib_dst->next_hop;
+
+    if (nib != NULL) {
+        nib->mode &= ~(_DC);
+    }
+    _nib_dst_clear(nib_dst);
+}
+
+static inline _nib_dst_t *_nib_pl_add(unsigned iface, const ipv6_addr_t *pfx,
+                                      unsigned pfx_len)
+{
+    _nib_dst_t *nib_dst = _nib_dst_alloc(NULL, iface, pfx, pfx_len);
+    if (nib_dst != NULL) {
+        nib_dst->next_hop->mode |= (_PL);
+    }
+    return nib_dst;
+}
+
+static inline void _nib_pl_remove(_nib_dst_t *nib_dst)
+{
+    _nib_t *nib = nib_dst->next_hop;
+
+    if (nib != NULL) {
+        nib->mode &= ~(_PL);
+    }
+    _nib_dst_clear(nib_dst);
+}
+
+static inline _nib_dst_t *_nib_ft_add(const ipv6_addr_t *next_hop, unsigned iface,
```

Comment:
this function looks very similar to `_nib_pl_add()`. What's your opinion on refactoring the common parts to `_nib_dst_add()` and just provide a macro or inline func that calls `_nib_dst_add()` with the right parameters?